TerriaJS/terriajs
# Chart fixes

Depends on https://github.com/TerriaJS/terriajs/pull/5760

### What this PR does

- Makes feature info panel chart labels more readable - https://github.com/TerriaJS/de-australia-map/issues/104
- Do not replace expanded charts if they have different titles - https://github.com/TerriaJS/de-australia-map/issues/105
- Fix broken chart data download button (commit https://github.com/TerriaJS/terriajs/pull/5795/commits/d3b48cb83ec9516510effe6d6c990fc74ed5ba64)
- Changed chart tooltip and legend bar implementation so that it can fit more legends gracefully. (https://github.com/TerriaJS/terriajs/pull/5795/commits/120e5269150510982456e241135772a7fb314389 https://github.com/TerriaJS/terriajs/pull/5795/commits/8a2f7a18c9ed543e8a48b1eb9390962a686f3752)
- Improve status message when feature info panel chart is loading
  Previously:
  
![image](https://user-images.githubusercontent.com/1430/131962959-5604349d-cdf1-4e1d-9353-98d19a5a1691.png)


  Now:

![image](https://user-images.githubusercontent.com/1430/131962874-51c64373-bf6d-4cd7-89d3-febd0474fa6d.png)

### Testing

- Open this [link](http://ci.terria.io/chart-fixes/#share=s-dShjXJ4zG9AUnTm2KAOHFFQVZ09&https://gist.githubusercontent.com/na9da/3c46309baf6cca7be9d301e6348eb75e/raw/e35b8ef8d21d9016a3ec8f23d0df5ab56d940532/coastal.json).
- Click a coastal point to see feature info panel with a chart
- Check out the x-axis labels (there should be enough labels to make the chart readable)
- Expand the chart
- Now click another point and expand chart again
- You should be able to see both charts together (previously only the new chart replaced the existing one).
- Clicking the download button should trigger download of the chart data without erroring.


### Checklist
